grails-app/services/org/pih/warehouse/inventory/InventoryService.groovy

```diff
@@ -3432,4 +3436,22 @@ class InventoryService implements ApplicationContextAware {
             }
         }
     }
+
+    Boolean areLotsValid(List<Object> items) {
+        return items?.every {
+            InventoryItem inventoryItem = InventoryItem.findByLotNumberAndProduct(it?.lotNumber, Product.findByProductCode(it?.productCode))
+
+            if (!inventoryItem) {
+                return true
+            }
+
+            Integer availableQuantity = productAvailabilityService.getQuantityOnHand(inventoryItem)
```

Comment:
This is confusing. Quantity on Hand != Quantity Available. Variable name should reflect the data we are working with.